src/Razor/src/Microsoft.AspNetCore.Razor.ExternalAccess.RoslynWorkspace/RazorWorkspaceListener.cs

```diff
@@ -150,30 +118,43 @@ private void EnqueueUpdate(Project? project)
         }
 
         var projectId = project.Id;
-        if (_workQueues.TryGetValue(projectId, out var scheduler))
+        _workQueue.AddWork(projectId);
+    }
+
+    private async ValueTask UpdateCurrentProjectsAsync(ImmutableArray<ProjectId> projectIds, CancellationToken cancellationToken)
+    {
+        if (_workspace is null)
         {
-            _logger.LogTrace("{projectId} scheduling task due to workspace event", projectId);
+            _logger?.LogTrace("No workspace available when trying to write project info");
+            return;
+        }
 
-            scheduler.ScheduleAsyncTask(ct => SerializeProjectAsync(projectId, ct), CancellationToken.None);
+        if (_projectInfoFileName is null)
+        {
+            _logger?.LogTrace("Don't know what filename to use");
+            return;
```

Comment:
Similar feedback as above. I'm not seeing how this could happen. Consider something like:

```C#
var projectInfoFileName = _projectInfoFileName.AssumeNotNull();
```